### PR TITLE
Add an interface for LLM runner

### PR DESCRIPTION
**Pull request by kirklandsign**
In case we have custom LLM runners other than llama runner, we want to have a uniform interface

### CODE REVIEW

File: extension/llm/runner/runner_interface.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#include <string>
+
+#include <executorch/extension/llm/runner/stats.h>
+#include <executorch/extension/module/module.h>
+
+namespace executorch {
+namespace extension {
+namespace llm {
+
+class ET_EXPERIMENTAL RunnerInterface {
```
Comment (larryliu0820) [inline extension/llm/runner/runner_interface.h:24]: Maybe call it `IRunner`? Also what is the actual usage of having this interface? Can you also include the changes you want to make, to reduce duplicate code?

Comment (kirklandsign) [reply extension/llm/runner/runner_interface.h:24]: > Maybe call it IRunner
Works for me

Actual usage: https://github.com/pytorch/executorch/pull/6208/files#diff-344496b5b78dc04421b5eddcb0f261269c3150e122a9be63676da565955cb4f7R123-R131

That code is not ready yet

File: extension/llm/runner/irunner.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+namespace executorch {
+namespace extension {
+namespace llm {
+
+class ET_EXPERIMENTAL IRunner {
+ public:
+  virtual ~IRunner() = default;
+
+  // Checks if the model is loaded.
+  virtual bool is_loaded() const = 0;
```
Comment (larryliu0820) [inline extension/llm/runner/irunner.h:29]: This can have a default implementation right?

Comment (kirklandsign) [reply extension/llm/runner/irunner.h:29]: Maybe not. 
https://github.com/pytorch/executorch/blob/516a417d7d6f6f0b079cfc5bc42518e15cd9e1e4/examples/models/llama/runner/runner.cpp#L62-L65
it's quite impl detail

**Review by larryliu0820 — APPROVED** [review]

**Comment by larryliu0820** [comment]
Please fix the CI failures

**Pull request closed without merging**